File: client/src/app/shared/models/product.ts
export interface IProduct {
    id: number;
    name: string;
    shortDescription: string;
    description: string;
    technicalDescription: string;
    price: number;
    pictureUrl: string;
    productType: string;
    productBrand: string;
    productSKU: number;
    numberOfRatings: number;
    productRatingScore: number;
    photos: IPhoto[];
}

export interface IPhoto {
  id: number;
  pictureUrl: string;
  fileName: string;
  isMain: boolean;
  publicId: string;
  }

export interface IProductToCreate {
  name: string;
  shortDescription: string;
  description: string;
  technicalDescription: string;
  price: number;
  pictureUrl: string;
  productTypeId: number;
  productBrandId: number;
  productSKU: number;
  numberOfRatings: number;
  productRatingScore: number;
}

export class ProductFormValues implements IProductToCreate {
  name = '';
  shortDescription = '';
  description = '';
  technicalDescription = '';
  price = 0;
  pictureUrl = '';
  productTypeId: number;
  productBrandId: number;
  productSKU: number;
  numberOfRatings: number;
  productRatingScore: number;
  file: File;

  constructor(init?: ProductFormValues) {
    Object.assign(this, init);
  }
}

File: Core/Entities/Product.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;

namespace Core.Entities
{
    public class Product : BaseEntity
    {
        public string Name { get; set; }

        public string ShortDescription { get; set; }
        public string Description { get; set; }

        public string TechnicalDescription { get; set; }
        public Decimal Price { get; set; }
        public ProductType ProductType { get; set; }

        public int ProductTypeId { get; set; }
        public ProductBrand ProductBrand { get; set; }
        public int ProductBrandId { get; set; }

        public int ProductSKU { get; set; }

        public int ProductRatingScore { get; set; }

        public int NumberOfRatings { get; set; }

        private readonly List<Photo> _photos = new List<Photo>();
        public IReadOnlyList<Photo> Photos => _photos.AsReadOnly();


        public void AddPhoto(string pictureUrl, string fileName, string publicId , bool isMain = false)
        {
            var photo = new Photo
            {
                FileName = fileName,
                PictureUrl = pictureUrl,
                PublicId = publicId
            };

            if (_photos.Count == 0)
            {
                photo.IsMain = true;
            }

            _photos.Add(photo);
        }

        public void RemovePhoto(int id)
        {
            var photo = _photos.Find(x => x.Id == id);
            _photos.Remove(photo);
        }

         public void SetMainPhoto(int id)
        {
            var currentMain = _photos.SingleOrDefault(item => item.IsMain);
            foreach (var item in _photos.Where(item => item.IsMain))
            {
                item.IsMain = false;
            }
            
            var photo = _photos.Find(x => x.Id == id);
            if (photo != null)
            {
                photo.IsMain = true;
                if (currentMain != null)
                {
                    currentMain.IsMain = false;
                } 
            }
        }
    }
}
